### PR TITLE
TDD with ruby

### PR DESCRIPTION
**Pull request by Sajeelzafar**
### In this project, the tests for the methods fizzbuzz, reverse, and factorial are designed and implemented. The methods were designed using TDD (Test Driven Development) approach.

### CODE REVIEW

File: classes/solver.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      'fizzbuzz'
+    elsif (num % 3).zero?
+      'fizz'
+    elsif (num % 5).zero?
+      'buzz'
+    else
+      num.to_s
+    end
+  end
+end
```
Comment (Mihndim2020) [inline classes/solver.rb]: According to the project [requirements](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/testing/tdd_project.md), " `Remember to start with tests for all of the methods described above. Your commit history should prove that you have used TDD.` ", please make sure you write the tests first before implementing the functions, and that should be reflected in your commit history.

**Review by Mihndim2020 — CHANGES_REQUESTED** [review]
### Hi Team, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to get this project approved, but you are almost there!
ALMOST-MINION

# Status: Requires Changes ♻️
_Check the comments under the review._


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/solver.rb (hunk 1 above)
Comment (Sajeelzafar) [reply classes/solver.rb]: @Mihndim2020 thank you for taking out time to review our project. Can you please suggest to us how to make the commit history now look like we developed the testing first? Should we reset the commits, git reset --hard, to start and make a new branch?

**Review by KayLemba — APPROVED** [review]
[# STATUS: APPROVED :green_circle: :clap: 

Hello @TEAM      , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

HIGHLIGHTS:

- [x]  Your repo is professional 
- [x] Descriptive P.R







 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]

**Pull request merged**